src/main/java/net/minecraftforge/fluids/ForgeFlowingFluid.java

```diff
@@ -0,0 +1,243 @@
+import net.minecraft.tags.FluidTags;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.BlockRenderLayer;
+import net.minecraft.util.Direction;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.IWorld;
+import net.minecraft.world.IWorldReader;
+
+import javax.annotation.Nullable;
+import java.util.function.Supplier;
+
+public abstract class ForgeFlowingFluid extends FlowingFluid
+{
+    private final Supplier<Fluid> flowing;
+    private final Supplier<Fluid> still;
+    @Nullable
+    private final Supplier<Item> bucket;
+    @Nullable
+    private final Supplier<FlowingFluidBlock> block;
+    private final FluidAttributes.Builder builder;
+    private final boolean canMultiply;
+    private final int slopeFindDistance;
+    private final int levelDecreasePerBlock;
+    private final float explosionResistance;
+    private final BlockRenderLayer renderLayer;
+    private final int tickRate;
+
+    protected ForgeFlowingFluid(Properties properties)
+    {
+        this.flowing = properties.flowing;
+        this.still = properties.still;
+        this.builder = properties.attributes;
+        this.canMultiply = properties.canMultiply;
+        this.bucket = properties.bucket;
+        this.block = properties.block;
+        this.slopeFindDistance = properties.slopeFindDistance;
+        this.levelDecreasePerBlock = properties.levelDecreasePerBlock;
+        this.explosionResistance = properties.explosionResistance;
+        this.renderLayer = properties.renderLayer;
+        this.tickRate = properties.tickRate;
+    }
+
+    @Override
+    public Fluid getFlowingFluid()
+    {
+        return flowing.get();
+    }
+
+    @Override
+    public Fluid getStillFluid()
+    {
+        return still.get();
+    }
+
+    @Override
+    protected boolean canSourcesMultiply()
+    {
+        return canMultiply;
+    }
+
+    @Override
+    protected void beforeReplacingBlock(IWorld worldIn, BlockPos pos, BlockState state)
+    {
+        TileEntity tileentity = state.getBlock().hasTileEntity() ? worldIn.getTileEntity(pos) : null;
+        Block.spawnDrops(state, worldIn.getWorld(), pos, tileentity);
+    }
+
+    @Override
+    protected int getSlopeFindDistance(IWorldReader worldIn)
+    {
+        return slopeFindDistance;
+    }
+
+    @Override
+    protected int getLevelDecreasePerBlock(IWorldReader worldIn)
+    {
+        return levelDecreasePerBlock;
+    }
+
+    @Override
+    public BlockRenderLayer getRenderLayer()
+    {
+        return renderLayer;
+    }
+
+    @Override
+    public Item getFilledBucket()
+    {
+        return bucket != null ? bucket.get() : Items.AIR;
+    }
+
+    @Override
+    protected boolean func_215665_a(IFluidState p_215665_1_, IBlockReader p_215665_2_, BlockPos p_215665_3_, Fluid p_215665_4_, Direction p_215665_5_)
+    {
+        return p_215665_5_ == Direction.DOWN && !p_215665_4_.isIn(FluidTags.WATER);
```

Comment:
Don't have my compiler open at the moment, so don't remember what this method is for overall given the naming, but is checking against water the correct thing to be doing here given its a custom fluid.

Reply:
TBH I meant to look at what this is used for, but I forgot, will check.

Reply:
Did my best. Settled on using isEquivalentTo as default. Mods can override if needed.